addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/TestInfoSessionIdURL.java

```diff
@@ -132,25 +135,36 @@ public void setParent(PassiveScanThread parent) {
     }
 
     /**
-     * Scan the request. Currently it does nothing.
+     * Scan the response. Currently it does nothing. TODO: This method should really
+     * scan the contents of the response to see if it is HTML, and if so, look for
+     * HREFs and SRC tags and check if the URLs in them contain session IDs. This
+     * would enable ZAP to detect Session IDs in URLs exactly where they are
+     * occurring, rather than simply detecting the symptom of the problem in URLs of
+     * requests.
      *
-     * @param msg the HTTP message
-     * @param id the id of the request
+     * @param msg    the HTTP message
+     * @param id     the id of the response
+     * @param source the source code of the response
      */
     @Override
-    public void scanHttpRequestSend(HttpMessage msg, int id) {
-        // do Nothing it's related to response managed
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        // do Nothing. All work currently is done scanning the request.
     }
 
+    // Pattern used only in scanHttpRequestSend() method below.
+    private static final Pattern PATHSESSIONIDPATTERN =
+            Pattern.compile(
+                    "(jsession|cf)id=[\\dA-Z]{" + SESSION_TOKEN_MIN_LENGTH + ",}",
```

Comment:
Was `cfid` also seen in the path? (The scanner was just checking `jsessionid`.)

Reply:
No, we just added it as another thing to check for. Ideally, I would think that the full list of session IDs checked for earlier would be checked for here as well. But if you want the cfid check removed for some reason, that's fine.

Reply:
I think this is just a hack used with `jsessionid`, so no need to check others.